reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/FundingSubTypeResource.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.nhs.tis.StringConverter;
+
+/**
+ * REST controller for managing FundingType.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/api")
+public class FundingSubTypeResource {
+
+  private static final String ENTITY_NAME = "fundingSubType";
+
+  private final FundingSubTypeMapper fundingSubTypeMapper;
+  private final FundingSubTypeServiceImpl fundingSubTypeService;
+  private final FundingSubTypeValidator fundingSubTypeValidator;
+
+  /**
+   * Constructor for FundingSubTypeResource.
+   *
+   * @param fundingSubTypeMapper the mapper to convert between entity and dto
+   * @param fundingSubTypeService the service to handle business logic
+   * @param fundingSubTypeValidator the validator to validate user data
+   */
+  public FundingSubTypeResource(FundingSubTypeMapper fundingSubTypeMapper,
+      FundingSubTypeServiceImpl fundingSubTypeService,
+      FundingSubTypeValidator fundingSubTypeValidator) {
+    this.fundingSubTypeMapper = fundingSubTypeMapper;
+    this.fundingSubTypeService = fundingSubTypeService;
+    this.fundingSubTypeValidator = fundingSubTypeValidator;
+  }
+
+  /**
+   * POST  /funding-sub-type : Create a new fundingType.
+   *
+   * @param fundingSubTypeDto the fundingTypeDTO to create
+   * @return the ResponseEntity with status 201 (Created) and with body the new fundingSubTypeDTO,
+   *     or with status 400 (Bad Request) if the fundingSubType has already an ID
+   * @throws URISyntaxException if the Location URI syntax is incorrect
+   */
+  @PostMapping("/funding-sub-type")
+  @PreAuthorize("hasAuthority('reference:add:modify:entities')")
+  public ResponseEntity<FundingSubTypeDto> createFundingSubType(
+      @Valid @RequestBody FundingSubTypeDto fundingSubTypeDto) throws URISyntaxException {
+    log.debug("REST request to save FundingType : {}", fundingSubTypeDto);
+    fundingSubTypeValidator.validate(fundingSubTypeDto);
+    if (fundingSubTypeDto.getUuid() != null) {
+      return ResponseEntity.badRequest().headers(HeaderUtil
+          .createFailureAlert(ENTITY_NAME, "idexists",
+              "A new fundingSubType cannot already have an uuid")).body(null);
+    }
+    FundingSubType fundingSubType = fundingSubTypeMapper.toEntity(fundingSubTypeDto);
+    fundingSubType = fundingSubTypeService.save(fundingSubType);
+    FundingSubType fundingSubTypeWithParentInfo = fundingSubTypeService.findById(
+        fundingSubType.getUuid()).get();
```

Comment:
With this line, after a POST request, the returned JSON body is as below:
```
{
    "uuid": "57ee3dbd-9448-46bc-8844-3ddf09bd6516",
    "code": "TEST",
    "label": "test",
    "status": "CURRENT",
    "fundingType": {
        "id": 15,
        "uuid": "65303230-3134-3336-2d33-3131362d3131",
        "code": "ACADEMIC_UNIVERSITY",
        "label": "Academic - University",
        "status": "CURRENT",
        "allowDetails": true
    }
}
```
But if I remove this line, the returned JSON body is:
```
{
    "uuid": "96df4c8c-3a07-4ec7-8742-7fa5075b64c7",
    "code": "TEST",
    "label": "test",
    "status": "CURRENT",
    "fundingType": {
        "id": 15,
        "uuid": null,
        "code": null,
        "label": null,
        "status": null,
        "allowDetails": false
    }
}
```

Reply:
Yep, I see I was mistaken about what is happening.
Can we have a quick chat about when the full funding type should be included as I'm wondering if it may well be more correct to override the default behaviour in the repository/service?

Reply:
I tested POST endpoint in `SiteResource` locally, and the nested entity `organizationType` has the same behaviour and for Site, frontend always sends full data in the nested dto. So to keep them consistent, I removed this necessary line.